core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java

```diff
@@ -76,24 +99,25 @@ public static Comparator getComparatorByDataType(DataType dataType) {
 
   static class DoubleComparator implements Comparator<Object> {
     @Override public int compare(Object key1, Object key2) {
-      double result = (double) key1 - (double) key2;
-      if (result < 0) {
+      double key1Double1 = (double)key1;
```

Comment:
There is a scenario is a variable is a negative one the Key1 - Key2 wont give proper output. Better to check greater or less than operator.